build.gradle.kts

```diff
@@ -0,0 +1,214 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+plugins {
+  id("java")
+  id("maven")
+  id("java-gradle-plugin")
+  id("net.researchgate.release") version "2.6.0"
+  id("com.github.sherter.google-java-format") version "0.8"
+  id("checkstyle")
+  id("jacoco")
+}
+
+repositories {
+  mavenCentral()
+}
+
+java {
+  sourceCompatibility = JavaVersion.VERSION_1_8
+  targetCompatibility = JavaVersion.VERSION_1_8
+}
+
+group = "com.google.cloud.tools"
+
+dependencies {
+  compile(localGroovy())
+  compile(gradleApi())
+  compile("com.google.cloud.tools:appengine-plugins-core:0.9.8")
+
+  testCompile("commons-io:commons-io:2.4")
+  testCompile("junit:junit:4.12")
+  testCompile("org.hamcrest:hamcrest-library:1.3")
+  testCompile("org.mockito:mockito-core:2.23.4")
+}
+
+
+tasks.wrapper {
+  gradleVersion = "6.9"
+}
+
+tasks.jar.configure {
+  manifest {
+    attributes(
+      mapOf(
+        "Implementation-Title" to project.name,
+        "Implementation-Version" to version,
+        "Built-By" to System.getProperty("user.name"),
+        "Built-Date" to Date(),
+        "Built-JDK" to System.getProperty("java.version"),
+        "Built-Gradle" to gradle.gradleVersion
+      )
+    )
+  }
+}
+
+/* TESTING */
+tasks.test.configure {
+  testLogging {
+    showStandardStreams = true
+    exceptionFormat = FULL
+  }
+}
+
+sourceSets {
+  create("integTest") {
+    compileClasspath += main.get().output
+    runtimeClasspath += main.get().output
+  }
+}
+
+configurations {
+  named("integTestCompile").get().extendsFrom(testCompileClasspath.get())
+  named("integTestRuntime").get().extendsFrom(testRuntimeClasspath.get())
+}
+
+tasks.register<Test>("integTest") {
+  testClassesDirs = sourceSets.getByName("integTest").output.classesDirs
+  classpath = sourceSets.getByName("integTest").runtimeClasspath
+  outputs.upToDateWhen { false }
+}
+/* TESTING */
+
+
+/* RELEASING */
+tasks.register<Jar>("sourceJar") {
+  from(sourceSets.main.get().allJava)
+  classifier = "sources"
+}
+
+tasks.register<Jar>("javadocJar") {
+  dependsOn(tasks.javadoc)
+  from(tasks.javadoc.map { it.destinationDir!! })
+  classifier = "javadoc"
+}
+
+project.afterEvaluate {
+  tasks.register("writePom") {
+    val outputFile = file("$buildDir/pom/${project.name}-${project.version}.pom")
+    outputs.file(outputFile)
+
+    doLast {
+      maven {
+        pom {
+          project {
+            withGroovyBuilder {
```

Comment:
There is a cleaner syntax for this that would be preferable.  Like:
https://github.com/jamesward/gha-central-test/blob/fbf486aab2b6d60632a8c2b078457be1a05e1a74/build.gradle.kts#L21

Reply:
Would you consider merging #398 then ? https://github.com/GoogleCloudPlatform/app-gradle-plugin/pull/398/files#diff-c0dfa6bc7a8685217f70a860145fbdf416d449eaff052fa28352c5cec1a98c06R252

Reply:
Looks like that is a draft but also it seems like the goal of this is to minimize the changes, then with another PR clean things up / make them more Kotlin-like.  In which case, I'm good with the Groovy style for now.

Reply:
And sorry, I should have read the description in this PR.  :)

Reply:
For the record, I'm happy if we close this PR and merge #398 directly. A lot of things have changed in publishing land lately.

Reply:
#398 does too much. Let's take this one step at a time.